core/src/main/java/org/elasticsearch/common/network/NetworkService.java

```diff
@@ -210,6 +211,7 @@ public InetAddress resolvePublishHostAddresses(String publishHosts[]) throws IOE
     }
 
     /** resolves a single host specification */
+    @SuppressForbidden(reason = "does DNS lookup when bind/publish host is specified as hostname")
```

Comment:
i don't think we should make code more complex for this reason. the method's name is `resolveInternal` and i think its clear that it does DNS resolution.